File: jscomp/test/literals.js
'use strict';


var setter_suffix = "#=";

var setter_suffix_len = setter_suffix.length;

var js_array_ctor = "Array";

var js_type_number = "number";

var js_type_string = "string";

var js_type_object = "object";

var js_undefined = "undefined";

var js_prop_length = "length";

var param = "param";

var partial_arg = "partial_arg";

var prim = "prim";

var tmp = "tmp";

var create = "create";

var app = "_";

var app_array = "app";

var runtime = "runtime";

var stdlib = "stdlib";

var imul = "imul";

var js_debugger = "js_debugger";

var js_pure_expr = "js_pure_expr";

var js_pure_stmt = "js_pure_stmt";

var js_unsafe_downgrade = "js_unsafe_downgrade";

var js_fn_run = "js_fn_run";

var js_method_run = "js_method_run";

var js_fn_method = "js_fn_method";

var js_fn_mk = "js_fn_mk";

var js_fn_runmethod = "js_fn_runmethod";

var bs_deriving = "bs.deriving";

var bs_deriving_dot = "bs.deriving.";

var bs_type = "bs.type";

var node_modules = "node_modules";

var node_modules_length = 12;

var package_json = "package.json";

var bsconfig_json = "bsconfig.json";

var build_ninja = "build.ninja";

var suffix_cmj = ".cmj";

var suffix_cmi = ".cmi";

var suffix_ml = ".ml";

var suffix_mlast = ".mlast";

var suffix_mliast = ".mliast";

var suffix_mll = ".mll";

var suffix_d = ".d";

var suffix_mlastd = ".mlast.d";

var suffix_mliastd = ".mliast.d";

var suffix_js = ".js";

var suffix_mli = ".mli";

var suffix_cmt = ".cmt";

var suffix_cmti = ".cmti";

var commonjs = "commonjs";

var amdjs = "amdjs";

var goog = "goog";

var es6 = "es6";

var es6_global = "es6-global";

var amdjs_global = "amdjs-global";

var unused_attribute = "Unused attribute ";

var dash_nostdlib = "-nostdlib";

exports.js_array_ctor       = js_array_ctor;
exports.js_type_number      = js_type_number;
exports.js_type_string      = js_type_string;
exports.js_type_object      = js_type_object;
exports.js_undefined        = js_undefined;
exports.js_prop_length      = js_prop_length;
exports.param               = param;
exports.partial_arg         = partial_arg;
exports.prim                = prim;
exports.tmp                 = tmp;
exports.create              = create;
exports.app                 = app;
exports.app_array           = app_array;
exports.runtime             = runtime;
exports.stdlib              = stdlib;
exports.imul                = imul;
exports.setter_suffix       = setter_suffix;
exports.setter_suffix_len   = setter_suffix_len;
exports.js_debugger         = js_debugger;
exports.js_pure_expr        = js_pure_expr;
exports.js_pure_stmt        = js_pure_stmt;
exports.js_unsafe_downgrade = js_unsafe_downgrade;
exports.js_fn_run           = js_fn_run;
exports.js_method_run       = js_method_run;
exports.js_fn_method        = js_fn_method;
exports.js_fn_mk            = js_fn_mk;
exports.js_fn_runmethod     = js_fn_runmethod;
exports.bs_deriving         = bs_deriving;
exports.bs_deriving_dot     = bs_deriving_dot;
exports.bs_type             = bs_type;
exports.node_modules        = node_modules;
exports.node_modules_length = node_modules_length;
exports.package_json        = package_json;
exports.bsconfig_json       = bsconfig_json;
exports.build_ninja         = build_ninja;
exports.suffix_cmj          = suffix_cmj;
exports.suffix_cmi          = suffix_cmi;
exports.suffix_ml           = suffix_ml;
exports.suffix_mlast        = suffix_mlast;
exports.suffix_mliast       = suffix_mliast;
exports.suffix_mll          = suffix_mll;
exports.suffix_d            = suffix_d;
exports.suffix_mlastd       = suffix_mlastd;
exports.suffix_mliastd      = suffix_mliastd;
exports.suffix_js           = suffix_js;
exports.suffix_mli          = suffix_mli;
exports.suffix_cmt          = suffix_cmt;
exports.suffix_cmti         = suffix_cmti;
exports.commonjs            = commonjs;
exports.amdjs               = amdjs;
exports.goog                = goog;
exports.es6                 = es6;
exports.es6_global          = es6_global;
exports.amdjs_global        = amdjs_global;
exports.unused_attribute    = unused_attribute;
exports.dash_nostdlib       = dash_nostdlib;
/* No side effect */

File: jscomp/ext/literals.ml
(* Copyright (C) 2015-2016 Bloomberg Finance L.P.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * In addition to the permissions granted to you by the LGPL, you may combine
 * or link a "work that uses the Library" with a publicly distributed version
 * of this file to produce a combined library or application, then distribute
 * that combined work under the terms of your choosing, with no requirement
 * to comply with the obligations normally placed on you by section 4 of the
 * LGPL version 3 (or the corresponding section of a later version of the LGPL
 * should you choose to use a later version).
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. *)

let js_array_ctor = "Array"

let js_type_number = "number"

let js_type_string = "string"

let js_type_object = "object"

let js_type_boolean = "boolean"

let js_undefined = "undefined"

let js_prop_length = "length"

let prim = "prim"

let param = "param"

let partial_arg = "partial_arg"

let tmp = "tmp"

let create = "create" (* {!Caml_exceptions.create}*)

let runtime = "runtime" (* runtime directory *)

let stdlib = "stdlib"

let imul = "imul" (* signed int32 mul *)

let setter_suffix = "#="

let setter_suffix_len = String.length setter_suffix

let debugger = "debugger"

let fn_run = "fn_run"

let method_run = "method_run"

let fn_method = "fn_method"

let fn_mk = "fn_mk"
(*let js_fn_runmethod = "js_fn_runmethod"*)

(** nodejs *)
let node_modules = "node_modules"

let node_modules_length = String.length "node_modules"

let package_json = "package.json"

let bsconfig_json = "bsconfig.json"

let rescript_json = "rescript.json"

let build_ninja = "build.ninja"

(* Name of the library file created for each external dependency. *)
let library_file = "lib"

let suffix_a = ".a"

let suffix_cmj = ".cmj"

let suffix_cmo = ".cmo"

let suffix_cma = ".cma"

let suffix_cmi = ".cmi"

let suffix_cmx = ".cmx"

let suffix_cmxa = ".cmxa"

let suffix_mll = ".mll"

let suffix_ml = ".ml"

let suffix_mli = ".mli"

let suffix_res = ".res"

let suffix_resi = ".resi"

let suffix_mlmap = ".mlmap"

let suffix_cmt = ".cmt"

let suffix_cmti = ".cmti"

let suffix_ast = ".ast"

let suffix_iast = ".iast"

let suffix_d = ".d"

let suffix_js = ".js"

let suffix_gen_js = ".gen.js"

let suffix_gen_tsx = ".gen.tsx"

let esmodule = "esmodule"

let commonjs = "commonjs"

let es6 = "es6"
[@@ocaml.deprecated "Will be removed in v12"]

let es6_global = "es6-global"
[@@ocaml.deprecated "Will be removed in v12"]

let unused_attribute = "Unused attribute "

(** Used when produce node compatible paths *)
let node_sep = "/"

let node_parent = ".."

let node_current = "."

let gentype_import1 = "genType.import"
let gentype_import2 = "gentype.import"

let bsbuild_cache = ".bsbuild"

let sourcedirs_meta = ".sourcedirs.json"

(* Note the build system should check the validity of filenames
   espeically, it should not contain '-'
*)
let ns_sep_char = '-'

let ns_sep = "-"

let exception_id = "RE_EXN_ID"

let polyvar_hash = "NAME"

let polyvar_value = "VAL"

let cons = "::"

let hd = "hd"

let tl = "tl"

let lazy_done = "LAZY_DONE"

let lazy_val = "VAL"

let pure = "@__PURE__"
